shared/constants/wallets.js

```diff
@@ -396,10 +396,12 @@ export const paymentToYourInfoAndCounterparty = (
     case 'none':
       // Need to guard check that sourceType is non-empty to handle the
       // case when p is the empty value.
-      if (p.sourceType && p.sourceType !== 'ownaccount') {
+      //
+      // Also, sourceType might be 'keybase' when offline.
```

Comment:
@mlsteele any idea why this happens? Pretty sure it comes from the core side.

Reply:
Looks like `fillOwnAccounts` marks the type as OWNACCOUNT when it can look up the account name, which doesn't happen when offline.

Incidentally, `sourceType` can also be `stellar` in this case, so I think I'll just remove the whole check.

Reply:
This should be pretty [predictable](https://github.com/keybase/client/blob/master/protocol/avdl/stellar1/local.avdl#L85) when online. Not sure what it does offline though. Do the screens that use this work offline? What do you see in PaymentLocal do you see? Or how can I repro this?

Reply:
I'm in the process of adding a screen to the wallets page when relevant RPCs fail to display a 'can't load, retry?' page, and that's how I encountered this. I don't think we'll make the wallets page work offline more than that anytime soon, though.

e.g., when online, for a particular transaction between accounts I see:

```
{
  "amountDescription": "0.1400000 XLM",
  "delta": 2,
  "fromAccountID": "GBRSHUWDHFU6OQG4UOCSO6PZZGFY4OCDWL7GQBWI677POUR2EMCYNSNG",
  "fromAccountName": "secondary",
  "fromType": 4,
  "fromUsername": "",
  "id": "9c3e21641314639915a207311acd58297f11b3eaaf0fe04f06e559f9e8fc4b6a",
  "issuerDescription": "",
  "note": "",
  "noteErr": "",
  "originalToAssertion": "",
  "showCancel": false,
  "statusDescription": "completed",
  "statusDetail": "",
  "statusSimplified": 3,
  "time": 1539842796000,
  "toAccountID": "GBEXUI5PJ6VVSJJMHCQTVCOTZT6QBHNUZ46EQBMJXMTNWPXZACBP7R7G",
  "toAccountName": "primary",
  "toAssertion": "",
  "toType": 4,
  "toUsername": "",
  "unread": false,
  "worth": "",
  "worthAtSendTime": ""
}
```

and then when I go offline I see

```
{
  "amountDescription": "0.1400000 XLM",
  "delta": 1,
  "fromAccountID": "GBRSHUWDHFU6OQG4UOCSO6PZZGFY4OCDWL7GQBWI677POUR2EMCYNSNG",
  "fromAccountName": "",
  "fromType": 2,
  "fromUsername": "",
  "id": "9c3e21641314639915a207311acd58297f11b3eaaf0fe04f06e559f9e8fc4b6a",
  "issuerDescription": "",
  "note": "",
  "noteErr": "",
  "originalToAssertion": "",
  "showCancel": false,
  "statusDescription": "completed",
  "statusDetail": "",
  "statusSimplified": 3,
  "time": 1539842796000,
  "toAccountID": "GBEXUI5PJ6VVSJJMHCQTVCOTZT6QBHNUZ46EQBMJXMTNWPXZACBP7R7G",
  "toAccountName": "",
  "toAssertion": "",
  "toType": 2,
  "toUsername": "",
  "unread": false,
  "worth": "",
  "worthAtSendTime": ""
}
```

(Note the change to `fromType` and `toType`.)

As for reproing, it might be difficult after the Reloadable component goes in, but here's how you'd see this from the frontend debugger:

1) Pick a particular transaction between accounts and copy its ID to the clipboard.
2) Set a breakpoint for `rpcPaymentResultToPaymentResult` in `constants/wallets.js`
3) Edit that breakpoint and add the condition `w.payment.id === 'your transaction ID'`
4) Navigate to the wallet for that transaction, note what `w.payment` is when you hit the breakpoint, and continue.
5) Turn your Wi-Fi off.
6) Navigate away, and back to the wallet. Note `w.payment` again when you hit the breakpoint.